src/consumer.ts

```diff
@@ -1,11 +1,34 @@
+import { Connection } from "./connection"
 import { Message } from "./producer"
 
 export type ConsumerFunc = (message: Message) => void
 
 export class Consumer {
-  constructor(readonly handle: ConsumerFunc, readonly consumerId: number) {}
+  private connection: Connection
+  private stream: string
+  public consumerId: number
+  public consumerRef: string
+
+  constructor(
+    readonly handle: ConsumerFunc,
+    params: {
+      connection: Connection
+      stream: string
+      consumerId: number
+      consumerRef: string
```

Comment:
Are you sure it's mandatory for the conusmerRef??

Reply:
the store and retrieval of offset is based on the consumerRef so I'll keep it mandatory, otherwise the functionality need to check is the ref is present before act (and maybe throw an exception???)

Reply:
I have checked the documentation in C# https://rabbitmq.github.io/rabbitmq-stream-dotnet-client/stable/htmlsingle/index.html#_creating_a_consumer and the consumerRef is not mandatory. It's used if you want to enable this feature ... https://rabbitmq.github.io/rabbitmq-stream-dotnet-client/stable/htmlsingle/index.html#consumer-offset-tracking So I suggest to implement it in the correct way. If you pass the consumerRef you can store the offset otherwise it should rise an exception.